desc/compute/_fast_ion.py

```diff
@@ -354,3 +374,453 @@ def fun(pitch_inv):
+    resolution_requirement="tz",
+    grid_requirement={"can_fft2": True},
+    **_bounce_doc,
+)
+@partial(
+    jit,
+    static_argnames=[
+        "Y_B",
+        "num_transit",
+        "num_well",
+        "num_quad",
+        "num_pitch",
+        "pitch_batch_size",
+        "surf_batch_size",
+        "spline",
+    ],
+)
+def _dJ_ds(params, transforms, profiles, data, **kwargs):
+    """The max-J term.
+
+    Bounce-averaged binormal drift.
+    Normalization has been chosen to eliminate dependence on
+    num_transits.
+    """
+    # noqa: unused dependency
+    theta = kwargs["theta"]
+    Y_B = kwargs.get("Y_B", theta.shape[-1] * 2)
+    alpha = kwargs.get("alpha", jnp.array([0.0]))
+    num_transit = kwargs.get("num_transit", 20)
+    num_pitch = kwargs.get("num_pitch", 64)
+    num_well = kwargs.get("num_well", Y_B * num_transit)
+    pitch_batch_size = kwargs.get("pitch_batch_size", None)
+    surf_batch_size = kwargs.get("surf_batch_size", 1)
+    assert (
+        surf_batch_size == 1 or pitch_batch_size is None
+    ), f"Expected pitch_batch_size to be None, got {pitch_batch_size}."
+    spline = kwargs.get("spline", True)
+    quad = (
+        kwargs["quad"]
+        if "quad" in kwargs
+        else get_quadrature(
+            leggauss(kwargs.get("num_quad", 32)),
+            (automorphism_sin, grad_automorphism_sin),
+        )
+    )
+
+    def dJ_ds0(data):
+        bounce = Bounce2D(
+            grid,
+            data,
+            data["theta"],
+            Y_B,
+            alpha,
+            num_transit,
+            quad,
+            is_fourier=True,
+            spline=spline,
+        )
+
+        def fun(pitch_inv):
+            poloidal_drift = bounce.integrate(
+                [_poloidal_drift],
+                pitch_inv,
+                data,
+                ["cvdrift (periodic)", "gbdrift (periodic)", "gbdrift (secular)/phi"],
+                bounce.points(pitch_inv, num_well),
+                is_fourier=True,
+            )
+
+            # Take sum over wells
+            dJ_ds = jnp.sum(poloidal_drift, axis=-1)
+
+            # max drift < 0 provides TEM(trapped electron mode)
+            # stability for all rhos and pitches
+            return dJ_ds
+
+        # Output dimension (rho, alpha, lambda)
+        return batch_map(fun, data["pitch_inv"], pitch_batch_size)
+
+    grid = transforms["grid"]
+    data["J_s"] = -1 * _compute(
+        dJ_ds0,
+        {
+            "cvdrift (periodic)": data["cvdrift (periodic)"],
+            "gbdrift (periodic)": data["gbdrift (periodic)"],
+            "gbdrift (secular)/phi": data["gbdrift (secular)/phi"],
+        },
+        data,
+        theta,
+        grid,
+        num_pitch,
+        surf_batch_size,
+    )
+
+    # Since there isn't a standard method of normalization, I choose this.
+    # Gives the right result (J_s < 0 everywhere) for Goodman's elongated QI.
+    data["J_s"] = data["J_s"] / jnp.max(jnp.abs(data["J_s"]))
```

Comment:
I'd prefer to move this sort of normalization to the relevant objective, and keep the compute qty as close to the "raw physics" as possible